### PR TITLE
Limit the rate at which new Rsync jobs are generated

### PR DESCRIPTION
**Pull request by mikewallace1979**
This is a fix for https://github.com/EnterpriseDB/barman/issues/334 where the number of new SSH connections created
by Barman's Rsync jobs can exceed the value of MaxStartups on the target server.

### CODE REVIEW

**Review by didiermichel — COMMENTED** [review]
Nice  feature. I did like the `_apply_rate_limit` function.
Also opened the discussion on `workers_start_rate` before I actually read the full documentation. So It is maybe not a must do as someone else would get it right the first time.

File: doc/barman.1.d/50-backup.md [hunk 1]
```diff
@@ -53,6 +53,16 @@ backup *SERVER_NAME*
         value of the parameter `parallel_jobs`, if present in the
         configuration file.
 
+    --jobs-start-rate
+    :   Maximum rate at which workers should be started. Overrides the value
```
[the remark this answers is not in the document]
Comment (mikewallace1979) [reply doc/barman.1.d/50-backup.md:57]: I found it hard coming up with anything that was even halfway satisfactory so I appreciate the input here!

Maybe:

```
--jobs-max-startups
--jobs-max-startups-seconds
```

and then updating the descriptions to state the effective rate is `jobs-max-startups / jobs-max-startups-seconds`?

[the remark this answers is not in the document]
Comment (didiermichel) [reply doc/barman.1.d/50-backup.md:57]: It does make more sense using those names

**Review by didiermichel — APPROVED** [review]

**Comment by didiermichel** [comment]
I like the batch-period and batch-size choice

**Pull request merged**